packages/ethers/src/client.ts

```diff
@@ -341,6 +339,35 @@ export class Web3Modal extends Web3ModalScaffold {
         const txReceipt = await txResponse.wait()
 
         return (txReceipt?.hash as `0x${string}`) || null
+      },
+
+      writeContract: async (data: WriteContractArgs) => {
+        const { chainId, provider, address } = EthersStoreUtil.state
+
+        if (!provider) {
+          throw new Error('connectionControllerClient:sendTransaction - provider is undefined')
+        }
+
+        if (!address) {
+          throw new Error('connectionControllerClient:sendTransaction - address is undefined')
+        }
+
+        const browserProvider = new BrowserProvider(provider, chainId)
+        const signer = new JsonRpcSigner(browserProvider, address)
+        const contract = new Contract(data.tokenAddress, erc20ABI, signer)
```

Comment:
we're still depending on the ABI here :| 
I think it'd be best for now to keep this a separate method like `sendERC20` so we can be as flexible as we want with implementation. 

@enesozturk how did you work around this with convert? I assume you need to execute contract writes as well?

Reply:
We can send the abi as argument?

Reply:
On the convert side, we don't do contract writing since we are using 1Inch's aggregator contract and send transaction with parameters of `from`, `to`, and a few others.